testsuite/legacy-testsuite/src/test/java/org/sonatype/nexus/testsuite/security/nexus4257/Nexus4257CookieVerificationIT.java

```diff
@@ -1,277 +0,0 @@
-   * cut the direct plugin class dependency from ITs.
-   */
-  public static final String NO_SESSION_HEADER = "X-Nexus-Session";
-
-  @BeforeClass
-  public static void setSecureTest() {
-    TestContainer.getInstance().getTestContext().setSecureTest(true);
-  }
-
-  @Test
-  public void testCookieForStateFullClient()
-      throws Exception
-  {
-    setAnonymousAccess(false);
-
-    TestContext context = TestContainer.getInstance().getTestContext();
-    String username = context.getAdminUsername();
-    String password = context.getPassword();
-    String url = this.getBaseNexusUrl() + "content/";
-    URI nexusBaseURI = new URI(url);
-
-    DefaultHttpClient httpClient = new DefaultHttpClient();
-    httpClient.getParams().setParameter(CoreProtocolPNames.USER_AGENT, "SomeUAThatWillMakeMeLookStateful/1.0");
-    final BasicHttpContext localcontext = new BasicHttpContext();
-    final HttpHost targetHost =
-        new HttpHost(nexusBaseURI.getHost(), nexusBaseURI.getPort(), nexusBaseURI.getScheme());
-    httpClient.getCredentialsProvider().setCredentials(
-        new AuthScope(targetHost.getHostName(), targetHost.getPort()),
-        new UsernamePasswordCredentials(username, password));
-    AuthCache authCache = new BasicAuthCache();
-    BasicScheme basicAuth = new BasicScheme();
-    authCache.put(targetHost, basicAuth);
-    localcontext.setAttribute(ClientContext.AUTH_CACHE, authCache);
-
-    // stateful clients must login first, since other rest urls create no sessions
-    String loginUrl = this.getBaseNexusUrl() + "service/local/authentication/login";
-    assertThat(executeAndRelease(httpClient, new HttpGet(loginUrl), localcontext), equalTo(200));
-
-    // after login check content but make sure only cookie is used
-    httpClient.getCredentialsProvider().clear();
-    HttpGet getMethod = new HttpGet(url);
-    assertThat(executeAndRelease(httpClient, getMethod, null), equalTo(200));
-    Cookie sessionCookie = this.getSessionCookie(httpClient.getCookieStore().getCookies());
-    assertThat("Session Cookie not set", sessionCookie, notNullValue());
-    httpClient.getCookieStore().clear(); // remove cookies
-
-    // do not set the cookie, expect failure
-    HttpGet failedGetMethod = new HttpGet(url);
-    assertThat(executeAndRelease(httpClient, failedGetMethod, null), equalTo(401));
-
-    // set the cookie expect a 200, If a cookie is set, and cannot be found on the server, the response will fail
-    // with a 401
-    httpClient.getCookieStore().addCookie(sessionCookie);
-    getMethod = new HttpGet(url);
-    assertThat(executeAndRelease(httpClient, getMethod, null), equalTo(200));
-  }
-
-  /**
-   * Tests that session cookies are not set for the list of known stateless clients.
-   */
-  @Test
-  public void testCookieForStateLessClient()
-      throws Exception
-  {
-    setAnonymousAccess(false);
-
-    String[] statelessUserAgents = {
```

Comment:
This test was written back in the day when Nexus looked at user-agent to determine if to create a session.

Now, Nexus has no special handling internally looking at user agent and deciding if to create a session or not.

If we keep this test (method) I suppose that it verifies if requests to /content do not create sessions.

Do you think it is worth keeping? It is hard to test that all but one resource does not create sessions, but the generic /content request could be worth it I suppose just as a sanity.


It only creates sessions now at the login resource.

So does this test add value? Perhaps if we believe